storey/flow.py

```diff
@@ -615,31 +615,38 @@ def __init__(
             self,
             max_events: Optional[int] = None,
             timeout_secs: Optional[int] = None,
-            group_by_key: bool = False,
-            key: Optional[Union[str, callable]] = None,
+            key: Optional[Union[str, Callable[[Event], str]]] = None,
             **kwargs,
     ):
         super().__init__(**kwargs)
 
         self._max_events: int = max_events
         self._timeout_secs = timeout_secs
-        self._group_by_key = group_by_key
 
         if self._timeout_secs is not None and self._timeout_secs <= 0:
             raise ValueError('Batch timeout cannot be 0 or negative')
 
+        self._extract_key: Optional[Callable[[Event], str]] = None
```

Comment:
Semantically better to set `self._extract_key = create_extact_key_func()` or something and have that method return lambda rather than initializing it here to None and then init it right after in another place